diff -Nur -x '*.orig' -x '*~' indicator-session-0.1.7/src/lock-helper.c indicator-session-0.1.7.new/src/lock-helper.c
--- indicator-session-0.1.7/src/lock-helper.c	2009-10-14 19:15:55.000000000 +0200
+++ indicator-session-0.1.7.new/src/lock-helper.c	2009-10-14 19:16:16.000000000 +0200
@@ -130,9 +130,11 @@
 gboolean
 will_lock_screen (void)
 {
+/*
 	if (gdm_auto_login) {
 		return FALSE;
 	}
+*/
 	if (is_guest) {
 		return FALSE;
 	}
