=== modified file 'src/users-service.c'
--- src/users-service.c	2009-09-30 02:03:36 +0000
+++ src/users-service.c	2009-10-05 20:38:14 +0000
@@ -342,7 +342,7 @@
       if (check_new_session ())
         {
           mi = dbusmenu_menuitem_new ();
-          dbusmenu_menuitem_property_set (mi, DBUSMENU_MENUITEM_PROP_LABEL, _("New Session..."));
+          dbusmenu_menuitem_property_set (mi, DBUSMENU_MENUITEM_PROP_LABEL, _("Switch User..."));
           dbusmenu_menuitem_child_append (root, mi);
           g_signal_connect (G_OBJECT (mi), DBUSMENU_MENUITEM_SIGNAL_ITEM_ACTIVATED, G_CALLBACK (activate_new_session), NULL);
         }

